src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1149,6 +1166,47 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
         }
 
 #nullable enable
+        internal ImmutableArray<string> GetCompilerFeatureRequiredFeatures(EntityHandle token)
+        {
+            List<AttributeInfo>? infos = FindTargetAttributes(token, AttributeDescription.CompilerFeatureRequiredAttribute);
+
+            if (infos == null)
+            {
+                return ImmutableArray<string>.Empty;
+            }
+
+            ArrayBuilder<string> builder = ArrayBuilder<string>.GetInstance();
+
+            foreach (var info in infos)
+            {
+                if (TryExtractStringAndNamedArgBoolFromAttribute(info.Handle, out string? featureName, out bool isOptional, out string? namedArg, out bool isField))
+                {
+                    if (namedArg != "IsOptional" || isField)
+                    {
+                        isOptional = false;
+                    }
+                }
+                else if (TryExtractStringValueFromAttribute(info.Handle, out featureName))
```

Comment:
>TryExtractStringValueFromAttribute [](http://example.com/codeflow?start=25&length=34)

Are we trying to crack the attribute for the second time just because there was no named argument? I think we should have a specialized routine to crack this attribute in one pass, regardless of the application shape.